packages/svelte-ux/src/lib/utils/date.test.ts

```diff
@@ -0,0 +1,152 @@
+    expect(formatDate(DATE, PeriodType.WeekSun, 'short')).equal('11/19 - 11/25');
+    expect(formatDate(DATE, PeriodType.WeekSun, 'long')).equal('11/19/2023 - 11/25/2023');
+    expect(formatDate(DATE, PeriodType.WeekMon, 'long')).equal('11/20/2023 - 11/26/2023');
+  });
+
+  it('should format date for PeriodType.Month', () => {
+    expect(formatDate(DATE, PeriodType.Month, 'short')).equal('Nov 2023');
+    expect(formatDate(DATE, PeriodType.Month, 'long')).equal('November 2023');
+  });
+
+  it('should format date for PeriodType.Quarter', () => {
+    expect(formatDate(DATE, PeriodType.Quarter, 'short')).equal('Oct - Dec 2023');
+    expect(formatDate(DATE, PeriodType.Quarter, 'long')).equal('October - December 2023');
+  });
+
+  it('should format date for PeriodType.CalendarYear', () => {
+    expect(formatDate(DATE, PeriodType.CalendarYear, 'short')).equal('23');
+    expect(formatDate(DATE, PeriodType.CalendarYear, 'long')).equal('2023');
+  });
+
+  it('should format date for PeriodType.FiscalYearOctober', () => {
+    expect(formatDate(DATE, PeriodType.FiscalYearOctober, 'short')).equal('24');
+    expect(formatDate(DATE, PeriodType.FiscalYearOctober, 'long')).equal('2024');
+  });
+
+  it('should format date for PeriodType.BiWeek1Sun', () => {
+    expect(formatDate(DATE, PeriodType.BiWeek1Sun, 'short')).equal('11/12 - 11/25');
+    expect(formatDate(DATE, PeriodType.BiWeek1Sun, 'long')).equal('11/12/2023 - 11/25/2023');
+  });
+
+  it('should format date for PeriodType.undefined', () => {
+    expect(formatDate(DATE, undefined, 'short')).equal('2023-11-21T00:00:00-04:00');
+    expect(formatDate(DATE, undefined, 'long')).equal('2023-11-21T00:00:00-04:00');
+  });
+});
+
+describe('utcToLocalDate()', () => {
+  it('in with offset -00 => local', () => {
+    const utcDate = '2023-11-21T00:00:00-00:00';
+    const localDate = utcToLocalDate(utcDate);
+    expect(localDate.toISOString()).equal('2023-11-21T04:00:00.000Z');
+  });
+
+  it('in without offset, the utc is already +4, to local: another +4', () => {
+    const utcDate = '2023-11-21T00:00:00';
+    const localDate = utcToLocalDate(utcDate);
+    expect(localDate.toISOString()).equal('2023-11-21T08:00:00.000Z');
+  });
+});
+
+describe('localToUtcDate()', () => {
+  it('in with offset -04 => UTC', () => {
+    const localDate = '2023-11-21T00:00:00-04:00';
+    const utcDate = localToUtcDate(localDate);
+    expect(utcDate.toISOString()).equal('2023-11-21T00:00:00.000Z');
+  });
+
+  it('in with offset -00 => UTC', () => {
+    const localDate = '2023-11-21T04:00:00-00:00';
+    const utcDate = localToUtcDate(localDate);
+    expect(utcDate.toISOString()).equal('2023-11-21T00:00:00.000Z');
+  });
+
+  it('in without offset == UTC', () => {
+    const localDate = '2023-11-21T04:00:00';
+    const utcDate = localToUtcDate(localDate);
+    expect(utcDate.toISOString()).equal('2023-11-21T04:00:00.000Z');
+  });
+});
+
+describe('getMonthDaysByWeek()', () => {
```

Comment:
Could you check if it's the behavior you expect? (It was strange before, so I refactored, and remove a dep!!! :) )

Reply:
I love the idea of removing a `dep` :).  

I believe this is off by 1 and should return `2023-10-29` - `2023-12-02` instead of `2023-10-30` - `2023-12-03`, unless I'm not following the timezone correctly.

This is used on the [Month](https://svelte-ux.techniq.dev/docs/components/Month/) component, IIRC

![image](https://github.com/techniq/svelte-ux/assets/177476/b06bf434-83f2-489a-b7d3-0c548fc6fa13)

Reply:
Perfecto!

You expect Sunday first day of the week in US? or monday?

Reply:
Sunday :), but that would be locale specific.

Reply:
Dates... I love it! But actually, I might have realized something! Something BIG! (Or maybe I'm just stupid haha).
Could you pull the branch and do me a screenshot of your month component?
(I will explain after :p)

Reply:
Forget it 🤣, It should be good now.